File: app/src/main/java/com/kotobyte/utils/DefaultConfiguration.java
package com.kotobyte.utils;

import android.content.Context;
import android.content.SharedPreferences;

import com.kotobyte.R;
import com.kotobyte.base.Configuration;

import java.io.File;

public class DefaultConfiguration implements Configuration {

    private static final String KEY_CURRENT_DICTIONARY_VERSION = "current_dictionary_version";

    private SharedPreferences mSharedPreferences;
    private String mDictionaryFileName;
    private File mDictionaryFilePath;
    private int mLatestDictionaryVersion;

    public DefaultConfiguration(Context context) {
        mSharedPreferences = context.getSharedPreferences(context.getPackageName(), 0);

        mDictionaryFileName = context.getString(R.string.dictionary_file_name);
        mDictionaryFilePath = new File(context.getFilesDir(), mDictionaryFileName);
        mLatestDictionaryVersion = context.getResources().getInteger(R.integer.dictionary_version);
    }

    @Override
    public String getDictionaryFileName() {
        return mDictionaryFileName;
    }

    @Override
    public File getDictionaryFilePath() {
        return mDictionaryFilePath;
    }

    @Override
    public int getLatestDictionaryVersion() {
        return mLatestDictionaryVersion;
    }

    @Override
    public int getCurrentDictionaryVersion() {
        return mSharedPreferences.getInt(KEY_CURRENT_DICTIONARY_VERSION, 0);
    }

    @Override
    public void clearCurrentDictionaryVersion() {

        mSharedPreferences.edit()
                .remove(KEY_CURRENT_DICTIONARY_VERSION)
                .apply();
    }

    @Override
    public void updateCurrentDictionaryVersionToLatest() {

        mSharedPreferences.edit()
                .putInt(KEY_CURRENT_DICTIONARY_VERSION, mLatestDictionaryVersion)
                .apply();
    }
}

File: app/src/main/java/com/kotobyte/utils/DefaultConfiguration.kt
package com.kotobyte.utils

import android.content.Context

import com.kotobyte.R
import com.kotobyte.base.Configuration

import java.io.File

class DefaultConfiguration(context: Context) : Configuration {

    private val sharedPreferences = context.getSharedPreferences(context.packageName, 0)

    override val dictionaryFileName: String = context.getString(R.string.dictionary_file_name)
    override val dictionaryFilePath: File = File(context.filesDir, dictionaryFileName)
    override val latestDictionaryVersion: Int = context.resources.getInteger(R.integer.dictionary_version)

    override var currentDictionaryVersion: Int
        get() = sharedPreferences.getInt(KEY_CURRENT_DICTIONARY_VERSION, 0)
        set(version) = sharedPreferences.edit().putInt(KEY_CURRENT_DICTIONARY_VERSION, version).apply()

    companion object {
        private val KEY_CURRENT_DICTIONARY_VERSION = "current_dictionary_version"
    }
}
